lib/krane/deploy_task.rb

```diff
@@ -1,12 +1,402 @@
 # frozen_string_literal: true
+require 'yaml'
+require 'shellwords'
+require 'tempfile'
+require 'fileutils'
 
-require 'krane/deprecated_deploy_task'
+require 'krane/common'
+require 'krane/concurrency'
+require 'krane/resource_cache'
+require 'krane/kubernetes_resource'
+%w(
+  custom_resource
+  cloudsql
+  config_map
+  deployment
+  ingress
+  persistent_volume_claim
+  pod
+  network_policy
+  service
+  pod_template
+  pod_disruption_budget
+  replica_set
+  service_account
+  daemon_set
+  resource_quota
+  stateful_set
+  cron_job
+  job
+  custom_resource_definition
+  horizontal_pod_autoscaler
+  secret
+).each do |subresource|
+  require "krane/kubernetes_resource/#{subresource}"
+end
+require 'krane/resource_watcher'
+require 'krane/kubectl'
+require 'krane/kubeclient_builder'
+require 'krane/ejson_secret_provisioner'
+require 'krane/renderer'
+require 'krane/cluster_resource_discovery'
+require 'krane/template_sets'
+require 'krane/deploy_task_config_validator'
+require 'krane/resource_deployer'
+require 'krane/concerns/template_reporting'
 
 module Krane
-  class DeployTask < Krane::DeprecatedDeployTask
-    def initialize(**args)
-      raise "Use Krane::DeployGlobalTask to deploy global resources" if args[:allow_globals]
-      super(args.merge(allow_globals: false))
```

Comment:
We need to preserve this functionality.

Reply:
Removed `allow_globals: true`. `KraneDeployTest::test_deploy_globals_fails` shows that we still raise in case of global resources being present.

Reply:
A bunch of tests are still breaking and are not super trivial to move over. I'll continue this tomorrow:

- [x] test_cr_deploys_without_rollout_conditions_when_none_present
- [x] test_cr_deploys_without_rollout_conditions_when_none_present_deprecated
- [x] test_cr_merging
- [x] test_cr_failure_with_default_rollout_conditions
- [x] test_cr_failure_with_arbitrary_rollout_conditions
- [x] test_cr_success_with_arbitrary_rollout_conditions
- [x] test_cr_success_with_default_rollout_conditions
- [x] test_cr_succes_with_default_rollout_conditions_deprecated_annotation
- [ ] test_custom_resources_predeployed_deprecated
- [ ] test_crd_pruning
- [ ] test_crd_pruning_deprecated
- [ ] test_custom_resources_predeployed
- [ ] test_crd_can_fail